From e41f1cb84ace4bcf853b059733a333e4e4ea8758 Mon Sep 17 00:00:00 2001
From: Felix <google@ix5.org>
Date: Sun, 30 Dec 2018 19:00:20 +0100
Subject: [PATCH] ld.config: Allow /vendor/odm paths in addition to /odm

The Sony Open Devices Project is moving towards mounting all proprietary
partitions under /vendor and /mnt. Thus, libs under /vendor/odm need to
be accessible in the same vein as those in /odm.

Change-Id: Ie459b00474aedd9469fce464f299acf26b787fb4
Signed-off-by: Felix <google@ix5.org>
---
 rootdir/etc/ld.config.txt | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/rootdir/etc/ld.config.txt b/rootdir/etc/ld.config.txt
index 42dc7abe7..3c688ea41 100644
--- a/rootdir/etc/ld.config.txt
+++ b/rootdir/etc/ld.config.txt
@@ -112,19 +112,25 @@ namespace.sphal.visible = true
 
 namespace.sphal.search.paths  = /odm/${LIB}
 namespace.sphal.search.paths += /vendor/${LIB}
+namespace.sphal.search.paths += /vendor/odm/${LIB}
 
 namespace.sphal.permitted.paths  = /odm/${LIB}
 namespace.sphal.permitted.paths += /vendor/${LIB}
+namespace.sphal.permitted.paths += /vendor/odm/${LIB}
 
 namespace.sphal.asan.search.paths  = /data/asan/odm/${LIB}
 namespace.sphal.asan.search.paths +=           /odm/${LIB}
 namespace.sphal.asan.search.paths += /data/asan/vendor/${LIB}
 namespace.sphal.asan.search.paths +=           /vendor/${LIB}
+namespace.sphal.asan.search.paths += /data/asan/vendor/odm/${LIB}
+namespace.sphal.asan.search.paths +=           /vendor/odm/${LIB}
 
 namespace.sphal.asan.permitted.paths  = /data/asan/odm/${LIB}
 namespace.sphal.asan.permitted.paths +=           /odm/${LIB}
 namespace.sphal.asan.permitted.paths += /data/asan/vendor/${LIB}
 namespace.sphal.asan.permitted.paths +=           /vendor/${LIB}
+namespace.sphal.asan.permitted.paths += /data/asan/vendor/odm/${LIB}
+namespace.sphal.asan.permitted.paths +=           /vendor/odm/${LIB}
 
 # Once in this namespace, access to libraries in /system/lib is restricted. Only
 # libs listed here can be used.
@@ -262,6 +268,7 @@ namespace.default.visible = true
 
 namespace.default.search.paths  = /odm/${LIB}
 namespace.default.search.paths += /vendor/${LIB}
+namespace.default.search.paths += /vendor/odm/${LIB}
 
 namespace.default.permitted.paths  = /odm
 namespace.default.permitted.paths += /vendor
@@ -270,6 +277,8 @@ namespace.default.asan.search.paths  = /data/asan/odm/${LIB}
 namespace.default.asan.search.paths +=           /odm/${LIB}
 namespace.default.asan.search.paths += /data/asan/vendor/${LIB}
 namespace.default.asan.search.paths +=           /vendor/${LIB}
+namespace.default.asan.search.paths += /data/asan/vendor/odm/${LIB}
+namespace.default.asan.search.paths +=           /vendor/odm/${LIB}
 
 namespace.default.asan.permitted.paths  = /data/asan/odm
 namespace.default.asan.permitted.paths +=           /odm
-- 
2.20.1

